Servers/WindowServer/WindowManager.cpp

```diff
@@ -140,6 +140,18 @@ bool WindowManager::set_resolution(int width, int height)
     ClientConnection::for_each_client([&](ClientConnection& client) {
         client.notify_about_new_screen_rect(Screen::the().rect());
     });
+    for_each_window([&](Window& window) {
```

Comment:
We can get away with not doing this if changing the screen resolution was not successful.

Also, we don't need to pass in a reference to anything in `for_each_window()` - we only depend on the `window` in this lambda.

I feel like this type of logic for recalculating the rect's of a window might belong to the Window class. It has more internal logic for this type of thing that WindowManager does. The Manager classes of WindowManager generally notify what they are managing about events occurring rather than handling those events themselves.